shpc/main/modules/__init__.py

```diff
@@ -301,6 +380,9 @@ def install(self, name, tag=None, **kwargs):
         module_dir = os.path.join(self.settings.module_base, uri, tag.name)
         subfolder = os.path.join(uri, tag.name)
         container_dir = self.container.container_dir(subfolder)
+
+        # Cut out early if symlink desired but no home
+        self.check_symlink(module_dir, symlink)
```

Comment:
Why is it a requirement ? `shpc` will silently create all the other directories if they don't exist, cf the line right below.

Reply:
This was more respecting the user than anything functionally important - e.g., I want them to explicitly define and create the directory before the client drops stuff into it.

Reply:
This function also does more than check for the base - it looks to see if a particular module is already symlinked (which could be from the same or different module) and gives the person to bail out.

Reply:
> This was more respecting the user than anything functionally important - e.g., I want them to explicitly define and create the directory before the client drops stuff into it. It forces them to think about where they want to put it if they haven't defined a base.

The same argument could be said of `module_base` and `container_base` ;)

I agree with checking whether a symlink with the same name/version exists. I think only the constraint that `symlink_base` must exist needs to go, cf https://github.com/muffato/singularity-hpc/commit/0dbb0b260baa5158d1cdd9e0d42667d8f2cba431

Reply:
yes that's fair! And with that comparison I think I agree with you.

Reply:
Added! https://github.com/singularityhub/singularity-hpc/pull/502/commits/10cf953d9c29b282781b0e50c10bd4b3ab7295ef

Reply:
Also @muffato please take a look here when you get a chance! https://github.com/singularityhub/singularity-hpc/pull/506